### PR TITLE
Simple Search Results

### PR DESCRIPTION
**Pull request by stdavis**
This is a start on showing a simple results grid.

### CODE REVIEW

File: functions/generateZip.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+function convert(geojsonPath, format) {
+  console.log(`converting ${geojsonPath} to ${format}...`);
+  const dataset = gdal.open(geojsonPath);
+  let driver;
+  let fileExtension;
+  switch (format) {
+    case downloadFormats.shapefile:
+      driver = gdal.drivers.get('ESRI Shapefile');
```
Comment (steveoh) [inline functions/generateZip.js:42]: **question**: are the dependencies required for this available to functions?

File: src/components/search-wizard/Wizard.jsx [hunk 2]
```diff
@@ -35,6 +38,13 @@ export default function SearchWizard() {
     }
   }, [queryLayersConfig]);
 
+  const generateZip = httpsCallable(useFunctions(), 'generate');
+  const generateZipMutation = useMutation({
+    mutationFn: async (data) => {
```
Comment (steveoh) [inline src/components/search-wizard/Wizard.jsx:43]: **thought**: i don't think you need to explicitly use an async function here?

```suggestion
    mutationFn: generateZip
```

File: src/components/search-wizard/Wizard.jsx [hunk 3]
```diff
@@ -81,15 +88,48 @@ export default function SearchWizard() {
...
+              Back
+            </Button>
+          </>
+        ) : null}
+        {state.matches('download') ? (
+          <>
+            <Button
+              color={Button.Colors.primary}
+              className="w-full"
+              size={Button.Sizes.xl}
```
Comment (steveoh) [inline src/components/search-wizard/Wizard.jsx:118]: **thought**: would all these ternary statements be simplified if you moved all of this to a parent component that takes the current state as a prop?

File: src/utah-design-system/Checkbox.jsx [hunk 4]
```diff
@@ -8,45 +8,47 @@ export default function Checkbox({
...
+            <RadixCheckbox.Indicator>
+              <Icon bold name="checkmark" label={label} size="xs" />
+            </RadixCheckbox.Indicator>
+          </RadixCheckbox.Root>
+        </div>
+        <label
+          htmlFor={name || label}
+          className={clsx(
+            'ml-1 cursor-pointer leading-5',
+            disabled && '!text-slate-300'
```
Comment (steveoh) [inline src/utah-design-system/Checkbox.jsx:46]: **suggestion**: you can use clsx with class properties like

```suggestion
			className={clsx('ml-1 cursor-pointer leading-5', 
           {'!text-slate-300': disabled})
```

**thought**: I think all of the important modifiers are identifying an issue that would be resolved using clsx in a way where you pass the common classes as a string and the modifications as object properties.

```jsx
className={clsx(
              'flex h-4 w-4 items-center justify-center rounded border',
               {
								 'border-slate-500 bg-white data-[state=checked]:bg-slate-500 data-[state=checked]:text-white': !disabled,
                  'border-slate-300 bg-slate-50 data-[state=checked]:bg-slate-300 data-[state=checked]:text-white': disabled
               }      
            )}
          >
```

File: src/utah-design-system/RadioGroup.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                'rounded-full p-1',
+                !item.disabled &&
+                  'group-hover:bg-slate-200 group-focus:bg-slate-200 group-active:bg-slate-200'
+              )}
+            >
+              <RadixRadioGroup.Item
+                className={clsx(
+                  'flex h-4 w-4 items-center justify-center rounded-full border border-slate-500 bg-white',
+                  item.disabled &&
+                    '!border-slate-300 !bg-slate-50 data-[state=checked]:!bg-slate-300 data-[state=checked]:!text-white'
```
Comment (steveoh) [inline src/utah-design-system/RadioGroup.jsx:36]: **suggestion**: i would prefer if we didn't use the important modifier ever.

File: src/utah-design-system/RadioGroup.jsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                    'after:border-4 after:border-slate-500'
+                  )}
+                />
+              </RadixRadioGroup.Item>
+            </div>
+            <label
+              htmlFor={id}
+              className={clsx(
+                'ml-1 cursor-pointer leading-5',
+                item.disabled && '!text-slate-300'
```
Comment (steveoh) [inline src/utah-design-system/RadioGroup.jsx:55]: **suggestion**: refactor to remove !

File: src/utah-design-system/Checkbox.jsx (hunk 4 above)
Comment (stdavis) [reply src/utah-design-system/Checkbox.jsx]: Resolved in dcdb5f79f0d04ca9f35c62e740fb7e4b3ebd5272.

BTW - twJoin [does not support objects](https://github.com/dcastil/tailwind-merge/discussions/137#discussioncomment-3481605).

File: src/utah-design-system/RadioGroup.jsx (hunk 5 above)
Comment (stdavis) [reply src/utah-design-system/RadioGroup.jsx]: Resolved in dcdb5f79f0d04ca9f35c62e740fb7e4b3ebd5272.

File: src/utah-design-system/RadioGroup.jsx (hunk 6 above)
Comment (stdavis) [reply src/utah-design-system/RadioGroup.jsx]: Resolved in dcdb5f79f0d04ca9f35c62e740fb7e4b3ebd5272.

File: functions/generateZip.js (hunk 1 above)
Comment (stdavis) [reply functions/generateZip.js]: Removed the function entirely.

File: src/components/search-wizard/Wizard.jsx (hunk 3 above)
Comment (stdavis) [reply src/components/search-wizard/Wizard.jsx]: I'm going to punt on this until I have the advanced filter working. Then I plan on taking a fresh look at this. There's got to be a better way.

**Comment by stdavis** [comment]
@nathankota The download is ready for a new look. It's completely different. It does not rely on a function but rather makes requests to the `createReplica` endpoint for each the feature services directly for the client. There are several tradeoffs for this approach. First, not every feature service has this endpoint enabled. We can't enable it on the mapserv service since the source data needs to be in an enterprise geodatabase. It does work for AGOL items that have this checkbox checked:

image

We do have that checked for all of the SGID items in AGOL. So every layer that is pushed to SGID should work. But there are many that are not. Here's a list:

Ground Water Protection Zones
Source Water Assessment Zones
Surface Water Protection Zones
Oil and Gas - Wells
Dry Cleaner Facilities
Drinking Water System Ratings
Municipal Separate Storm Sewer System (MS4) Permits
Air Toxics Respiratory Hazard EJ Index
Air Toxics Cancer Risk EJ Index
Ozone EJ Index
Particulate Matter 2.5 EJ Index
Diesel Particulate Matter EJ Index
Traffic Proximity EJ Index
Lead Paint in Homes EJ Index
Superfund Proximity EJ Index
Risk Management Plan Facility Proximity EJ Index
Hazardous Waste Proximity EJ Index
Underground Storage Tanks EJ Index
Wastewater Discharge EJ Index
Ozone Nonattainment/Maintenance Areas
PM 2.5 Nonattainment/Maintenance Areas
EJ Demographic Index

For these, we will have to either convince the owners of the source feature services to turn on the export function (mostly the EJ Index layers) or we will have to push them to AGOL ourselves where we can turn it on. I think that both options are doable.

You can still perform searches on these layers, but the download checkboxes will be greyed out:
image

I'd love to know what you think about this newer solution.

File: src/components/search-wizard/Download.jsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        const layerName = searchLayer[fieldNames.queryLayers.layerName];
+
+        return (
+          <Checkbox
+            key={uniqueId}
+            checked={
+              searchLayer.supportsExport && selectedLayers.includes(uniqueId)
+            }
+            label={`${layerName}${
+              searchLayer.supportsExport ? '' : ' (download not available)'
```
Comment (steveoh) [inline src/components/search-wizard/Download.jsx:70]: **suggestion**: update language
```suggestion
              searchLayer.supportsExport ? '' : ' (download not supported)'
```

**Review by steveoh — APPROVED** [review]

File: functions/package.json [hunk 8]
```diff
@@ -14,10 +14,13 @@
...
   "dependencies": {
     "@google-cloud/secret-manager": "^4.2.2",
-    "firebase-admin": "^11.8.0",
+    "firebase-admin": "^11.9.0",
     "firebase-functions": "^4.4.0",
+    "gdal-async": "^3.7.0",
     "googleapis": "^118.0.0",
-    "yup": "^1.1.1"
+    "got": "^13.0.0",
+    "jszip": "^3.10.1",
```
[the remark this answers is not in the document]
Comment (stdavis) [reply functions/package.json]: b7f26b984a4d91e81fe65c40c47616f8c271a669

File: src/utah-design-system/RadioGroup.jsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+                    'after:rounded-full after:content-[""]',
+                    'after:border-4 after:border-slate-500'
+                  )}
+                />
+              </RadixRadioGroup.Item>
+            </div>
+            <label
+              htmlFor={id}
+              className={twJoin(
+                'ml-1 cursor-pointer leading-5',
```
[the remark this answers is not in the document]
Comment (stdavis) [reply src/utah-design-system/RadioGroup.jsx]: cc604a199efb2944ab8ab0313a7d0ae70304e5ec

**Comment by stdavis** [comment]
@nathankota Do you have any major concerns related to [my previous comment](https://github.com/agrc/deq-enviro/pull/520#issuecomment-1574293371) before I merge this?

**Comment by nathankota** [comment]
> @nathankota Do you have any major concerns related to [my previous comment](https://github.com/agrc/deq-enviro/pull/520#issuecomment-1574293371) before I merge this?

No major concerns from me, and I like the strategy. Should I start trying to get the data stewards (looks like Lexie, Diedre, and not sure who else) to turn on the export feature on AGOL services? Then we can take a look at the list again and decide what to do with the remaining layers? Limiting the quantity of layers we have to bring into SGID AGOL seems best.

**Comment by stdavis** [comment]
> Should I start trying to get the data stewards (looks like Lexie, Diedre, and not sure who else) to turn on the export feature on AGOL services?

Yes! That would be awesome. Thanks! You can check out the source feature service URLs in the [text config spreadsheet](https://docs.google.com/spreadsheets/d/1aVJ68hOyp4H6sKEEuL-xtB2qE_y6W0gDg35TgUSxtFg/edit#gid=0).

**Pull request merged**